### ISSUE TITLE
Add option to terminate search on first match

### ISSUE THREAD

**Issue by bonanza123** [gahag/bgrep#5]
An option like greps "-m 1" option would be nice.

**Comment by gahag** [comment]
Sounds like a great idea! Pull requests welcome.

**Comment by aquac** [comment]
I guess if we have no additional switches enabled, in particular not `-b` and/or `-v`, then we have this case because it relies on `is_match` (and not `find_iter`)

**Comment by gahag** [comment]
Indeed, but it would still be nice to have a flag like grep's `-m`, which allows to limit to an arbitrary number of occurrences.
